packages/testing/src/chai-retry-plugin/types.ts

```diff
@@ -0,0 +1,48 @@
+import type { Assertion } from 'chai';
+import { assertionPropertyKeys } from './constants';
+
+// Type of `this` in `addMethod` function
+export type AssertionStatic = typeof Assertion;
+export type AssertionMethod = (...args: unknown[]) => AssertionStatic;
+
+// Function provided as argument of `expect`
+export type FunctionToRetry = (...args: unknown[]) => unknown | Promise<unknown>;
+
+// Assertions gathered in a stack to re-assert the provided function's results
+export type AssertionStackItem = { method?: AssertionMethod; args?: unknown[]; isNegate?: boolean; key?: string };
```

Comment:
you can't have `method` and `key` simultaneously in the assertion item, right?
the same with `isNegate`.

Also it's not clear what the `key` is

Reply:
@yurii-ve fixed